File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/digitaltwin/GetPropertySubElementsRequest.java
package org.opcfoundation.webserver.service.message.digitaltwin;

import org.opcfoundation.webserver.types.digitaltwin.ObjectServiceContext;

public class GetPropertySubElementsRequest {
    private final String id;
    private final String propertyId;
    private final ObjectServiceContext context;

    public GetPropertySubElementsRequest(
            ObjectServiceContext context,
            String propertyId)
    {
        this.id = context.getObjectId().getId();
        this.propertyId = propertyId;
        this.context = context;
    }

    public String getId() {
        return id;
    }

    public String getPropertyId() {
        return propertyId;
    }

    public ObjectServiceContext getContext() {
        return context;
    }
}

File: opcua-webservice-node/src/server/service/message/digitialtwin/GetPropertySubElementsRequest.ts
import { ObjectServiceContext } from "../../../types/digitaltwin/ObjectServiceContext";

export class GetPropertySubElementsRequest {
    private readonly _id: string;
    private readonly _propertyId: string;
    private readonly _context: ObjectServiceContext;

    constructor(
        context: ObjectServiceContext,
        propertyId: string) {
        this._id = context.objectId.id;
        this._propertyId = propertyId;
        this._context = context;
    }

    get id(): string {
        return this._id;
    }

    get propertyId(): string {
        return this._propertyId;
    }

    get context(): ObjectServiceContext {
        return this._context;
    }
}